crates/node/builder/src/components/network.rs

```diff
@@ -2,39 +2,47 @@
 
 use std::future::Future;
 
-use reth_network::{NetworkHandle, NetworkPrimitives};
+use reth_network_api::FullNetwork;
+use reth_network_p2p::BlockClient;
+use reth_node_api::{BlockTy, HeaderTy};
 use reth_transaction_pool::TransactionPool;
 
 use crate::{BuilderContext, FullNodeTypes};
 
 /// A type that knows how to build the network implementation.
 pub trait NetworkBuilder<Node: FullNodeTypes, Pool: TransactionPool>: Send {
-    /// The primitive types to use for the network.
-    type Primitives: NetworkPrimitives;
+    /// The network built.
+    type Network: FullNetwork<
+        BlockHeader = HeaderTy<Node::Types>,
+        Client: BlockClient<Block = BlockTy<Node::Types>>,
+    >;
```

Comment:
I see, this is the actual change

this is kinda nice because this removes the NetworkHandle lock in.

there's perhaps an argument for not doing `FullNetwork: NetworkPrimitives` and instead make this an associated type, but since NetworkHandle etc, are generic over that we can just add the impl to NetworkHandle directly.
I think this works.